protostar/configuration_file/configuration_file.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import List, Optional, Union
+
+from typing_extensions import Protocol
+
+from protostar.protostar_exception import ProtostarException
+from protostar.utils.protostar_directory import VersionType
+
+PrimitiveTypesSupportedByConfigurationFile = Union[str, int, bool]
+
+
+class ConfigurationFile(Protocol):
+    def get_min_protostar_version(self) -> Optional[VersionType]:
+        ...
+
+    def get_contract_names(self) -> List[str]:
+        ...
+
+    def get_contract_source_paths(self, contract_name: str) -> List[Path]:
+        ...
+
+    def get_lib_path(self) -> Optional[Path]:
+        ...
+
+    def get_command_argument(
+        self, command_name: str, argument_name: str, profile_name: Optional[str] = None
+    ) -> Optional[
+        Union[
+            PrimitiveTypesSupportedByConfigurationFile,
+            List[PrimitiveTypesSupportedByConfigurationFile],
+        ]
+    ]:
+        ...
```

Comment:
wouldn't it be easier to offload the role of filtering to call-site and let this method simply return everything it knows about all commands?

seems like you will end up duplicating filtering in `V2` struct

Reply:
> wouldn't it be easier to offload the role of filtering to call-site and let this method simply return everything it knows about all commands?

Nah, the caller (`ArgumentValueFromConfigProvider`) needs only argument value.

The code won't be duplicated in V2 because redesigned protostar.toml won't use `protostar` prefix.

Reply:
I don't understand.

`ArgumentValueFromConfigProvider` does accept command name and argument name and returns argument value. This entity can easily filter the expected value from a dictionary `dict[tuple[CommandName, ArgumentName], ArgumentValue]`, and I would expect `ConfigurationFile` to simply return such dictionary.

Reply:
This change will complicate the `ConfigurationFile` and `ArgumentValueFromConfigProvider`. `ConfigurationFile` still needs to expose `get_attribute` method.